### PR TITLE
No action in ensureRoleBinding if no change needs to be made;

### PR DESCRIPTION
**Pull request by ycliuhw**
This PR makes the role binding update faster if no update needs to be done.

### CODE REVIEW

File: caas/kubernetes/provider/rbac.go [hunk 1]
```diff
@@ -482,6 +499,9 @@ func (k *kubernetesClient) ensureRoleBinding(rb *rbacv1.RoleBinding) (out *rbacv
 		if v.GetName() == rb.GetName() {
 			name := v.GetName()
 			UID := v.GetUID()
+			if isRoleBindingEqual(v, *rb) {
```
Comment (wallyworld) [inline caas/kubernetes/provider/rbac.go:502]: Do we really need to list them (above) based on labels? Would it be more efficient to get any existing rb by name and update if found, or create if not?

**Review by wallyworld — APPROVED** [review]

File: caas/kubernetes/provider/rbac.go (hunk 1 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/rbac.go:502]: I don't see a huge difference between list and get here.
I just had a check, by changing this, I need to update lots tests' gomock which is painful.

Comment (ycliuhw) [reply caas/kubernetes/provider/rbac.go]: ok, I changed my mind.

**Pull request merged**